### PR TITLE
refactor(stream): change the encoding of stream consumer group

### PR DESCRIPTION
**Pull request by Yangsx-1**
The encoding way of stream consumer group is not efficient, this PR changes the encoding according to this [discussion](https://github.com/apache/kvrocks-website/pull/222#issuecomment-2130783692).

### CODE REVIEW

**Review by PragmaTwice — DISMISSED** [review]

**Comment by PragmaTwice** [comment]
Should `StreamEntryID::Maximum` be `{UINT64_MAX - 1, ..}` to avoid the `-1` case here?

**Comment by Yangsx-1** [comment]
Can we merge this PR?

**Comment by git-hulk** [comment]
@Yangsx-1 You can merge PR once the CI passed

**Review by git-hulk — DISMISSED** [review]

**Comment by PragmaTwice** [comment]
There is one issue that needs to be considered: after the modification, the case where `EID MS` is -1 needs to be excluded when traversing (iterating) all sub-keys (`EID MS|EID SEQ`).

As I mentioned [here](https://github.com/apache/kvrocks/pull/2384#issuecomment-2200221083).

**Comment by Yangsx-1** [comment]
> There is one issue that needs to be considered: after the modification, the case where `EID MS` is -1 needs to be excluded when traversing (iterating) all sub-keys (`EID MS|EID SEQ`).
> 
> As I mentioned [here](https://github.com/apache/kvrocks/pull/2384#issuecomment-2200221083).

It's been changed now.

**Review by git-hulk — DISMISSED** [review]

**Review by PragmaTwice — DISMISSED** [review]

File: tests/gocase/unit/type/stream/stream_test.go [hunk 1]
```diff
@@ -306,7 +306,7 @@ var streamTests = func(t *testing.T, enabledRESP3 string) {
 		require.Len(t, rdb.XRange(ctx, "vipstream", "(1-0", "(42-42").Val(), 1)
 		require.ErrorContains(t, rdb.XRange(ctx, "vipstream", "(-", "+").Err(), "ERR")
 		require.ErrorContains(t, rdb.XRange(ctx, "vipstream", "-", "(+").Err(), "ERR")
-		require.ErrorContains(t, rdb.XRange(ctx, "vipstream", "(18446744073709551615-18446744073709551615", "+").Err(), "ERR")
+		require.ErrorContains(t, rdb.XRange(ctx, "vipstream", "(18446744073709551614-18446744073709551615", "+").Err(), "ERR")
```
Comment (PragmaTwice) [inline tests/gocase/unit/type/stream/stream_test.go:309]: What will happen if it's `18446744073709551615` instead of `18446744073709551614` here?

Comment (Yangsx-1) [reply tests/gocase/unit/type/stream/stream_test.go:309]: There would be a nil answer rather than an error.

**Comment by PragmaTwice** [comment]
Hi @torwig, could you please review this PR when you have some time?

**Comment by torwig** [comment]
@PragmaTwice Sure. I'll review it today.

**Comment by torwig** [comment]
@Yangsx-1 @PragmaTwice Since we have `UINT64_MAX` as a delimiter, should it be an invalid `seq` of the stream entry ID?
Should we consider changing code [here](https://github.com/apache/kvrocks/blob/unstable/src/types/redis_stream_base.cc#L40), [here](https://github.com/apache/kvrocks/blob/unstable/src/types/redis_stream_base.cc#L182) and [here](https://github.com/apache/kvrocks/blob/unstable/src/types/redis_stream_base.cc#L219)?

**Comment by Yangsx-1** [comment]
> @Yangsx-1 @PragmaTwice Since we have `UINT64_MAX` as a delimiter, should it be an invalid `seq` of the stream entry ID? Should we consider changing code [here](https://github.com/apache/kvrocks/blob/unstable/src/types/redis_stream_base.cc#L40), [here](https://github.com/apache/kvrocks/blob/unstable/src/types/redis_stream_base.cc#L182) and [here](https://github.com/apache/kvrocks/blob/unstable/src/types/redis_stream_base.cc#L219)?

Yes, i'll change these code later.

**Review by torwig — APPROVED** [review]

**Review by PragmaTwice — APPROVED** [review]

**Pull request merged**